From: SeongJae Park <sj@kernel.org>
Date: Tue, 10 Feb 2026 21:38:34 -0800
Subject: [PATCH] ==== strictly respect min_nr_regions ====

mm/damon: strictly respect min_nr_regions

DAMON repsect min_nr_regions parameter by setting maximum size of
each region as total monitoring region size divided by the parameter
value.  And the limit is applied by preventing merge of regions that
result in a region larger than the maximum size.  There is no such
mechanism in split side.  If the current number of regions is smaller
than the min_nr_regions, therefore, it takes time until the number of
regions become equal to or larger than the min_nr_regions.

This makes fixed granularity monitoring difficult, especially when the
aggregation interval is long.  There was actually a report [1] of the
case.

More strictly respect min_nr_regions, by splitting regions larger than
the size.

Also, the limit is updted only after ops update, which is usually be
used for updating the

[1] https://lore.kernel.org/CAC5umyjmJE9SBqjbetZZecpY54bHpn2AvCGNv3aF6J=1cfoPXQ@mail.gmail.com

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus__wc7m_hd | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus__wc7m_hd

diff --git a/hkml_cv_bogus/hkml_cv_bogus__wc7m_hd b/hkml_cv_bogus/hkml_cv_bogus__wc7m_hd
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

